### ISSUE TITLE
Docker images for versions 0.7.3 and 0.7.4 do not run with the default entrypoint

### ISSUE THREAD

**Issue by tlkamp** [promregator/promregator#169]
## Environment
macOS Mojave
Docker version 19.03.5
Docker container versions 0.7.3 and 0.7.4

## Summary / Problem Statement

Using the default entrypoint for the 0.7.3 and 0.7.4 images prevents the JVM from starting due to an incorrect Java version.

## Observed Behavior

Trying to run the provided image with no modifications results in an error:

```console
$ docker run --rm promregator/promregator:0.7.3
+ '[' '' == '' ']'
+ JAVACMD=/usr/bin/java
+ '[' '' == '' ']'
+ JAVA_MEM_OPTS='-XX:+PrintFlagsFinal -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap -Xss600k -XX:ReservedCodeCacheSize=256m -XX:MaxMetaspaceSize=300m'
+ '[' '' == '' ']'
+ PROMREGATOR_CONFIG_DIR=/etc/promregator
+ cd /home/promregator
+ ln -sf ../..//etc/promregator/promregator.yml .
+ ln -sf ../..//etc/promregator/promregator.properties .
+ '[' -d /run/secrets ']'
+ set +x
+ /usr/bin/java -XX:+PrintFlagsFinal -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap -Xss600k -XX:ReservedCodeCacheSize=256m -XX:MaxMetaspaceSize=300m -Dspring.config.name=promregator -jar /opt/promregator/promregator.jar
Unrecognized VM option 'UseCGroupMemoryLimitForHeap'
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
```

The error is exactly the same in the 0.7.4 image.

## Expected Behavior

Promregator starts successfully.

## Additional info:
It looks like Promregator is expecting `/usr/bin/java` to be Java 1.8 but it is actually 1.11:

```console
$ docker run --rm promregator/promregator:0.7.3 /usr/bin/java -version
openjdk version "11.0.6" 2020-01-14
OpenJDK Runtime Environment (build 11.0.6+10-post-Debian-1deb10u1)
OpenJDK 64-Bit Server VM (build 11.0.6+10-post-Debian-1deb10u1, mixed mode, sharing)
```

```console
$ docker run --rm promregator/promregator:0.7.3 which java
/usr/local/openjdk-8/bin/java

$ docker run --rm promregator/promregator:0.7.3 java -version
openjdk version "1.8.0_242"
OpenJDK Runtime Environment (build 1.8.0_242-b08)
OpenJDK 64-Bit Server VM (build 25.242-b08, mixed mode)
```

I can get Promregator to run by overriding the `JAVACMD` environment variable used by `promregator.sh`:

```console 
$ docker run --rm -e JAVACMD=/usr/local/openjdk-8/bin/java promregator/promregator:0.7.3
```

I would prefer to not have to override the environment variable.

**Comment by eaglerainbow** [comment]
Hello, 
Welcome to Promregator! And thanks sharing this issue with us!

This looks totally odd and should not have happened. I'll have a look into it, as soon as I can.

**Comment by eaglerainbow** [comment]
Found the culprit: https://github.com/promregator/promregator/blob/6defd766cfc9f5e01557c07eedda2fe3d78d8e82/docker/Dockerfile#L18 installs 

* openjdk-11-jre-headless
* default-jre-headless 
* java-common

in the meantime implicitly.

Now, we only have to think about it what is the cure...

**Comment by eaglerainbow** [comment]
Too bad... We cannot 

```bash
apt-mark hold openjdk-11-jre-headless default-jre-headless java-common
``` 
as installing `ca-certificates-java` then results in 
``` 
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 ca-certificates-java : Depends: default-jre-headless but it is not going to be installed or
                                 java8-runtime-headless
E: Unable to correct problems, you have held broken packages.
```

**Comment by eaglerainbow** [comment]
seems like `ca-certificates` already supports generating `cacerts` natively in the meantime (see also https://github.com/docker-library/openjdk/issues/327 )

**Comment by eaglerainbow** [comment]
Manually checked that https certificates still work after f52fc53 (against IBM bluemix).

**Comment by eaglerainbow** [comment]
Version 0.7.5 is on the way...

**Comment by eaglerainbow** [comment]
[Version 0.7.5](https://github.com/promregator/promregator/releases/tag/v0.7.5) is released which should contain the fix.
Will also be solved with 0.8.0 and higher versions.

@tlkamp Thanks again for reporting this issue!
In case this did not solve it, please feel free to raise your hand and reopen this issue again.

**Comment by tlkamp** [comment]
@eaglerainbow thanks so much for addressing the issue so quickly! Very appreciated.
